ocaml/xapi/xapi_xenops.ml

```diff
@@ -647,29 +647,25 @@ module MD = struct
     let pci_address = Db.PCI.get_pci_id ~__context ~self:pci in
     Xenops_interface.Pci.address_of_string pci_address
 
+  let get_virtual_pci_address ~__context vgpu =
+   let device = vgpu.Db_actions.vGPU_device in
+   let virtual_pci_address  =  Xapi_globs.vgpu_pci_prefix ^ device ^".0" in
```

Comment:
Does this rely on the device always being plugged into slot 0? Because it's in the prefix of the address. What will we do in the cases of multiple vgpus, or we want to access a vgpu in another slot?

Reply:
Yes, since it's virtual PCI address which xapi defines here and pass them to DEMU.
Previously, when only one vgpu allowed, the virtual PCI address is defined in DEMU with slot always set 0. So after discussion, we decided to define virtual PCI address by xapi now and DEMU will handle the slot number.

Reply:
Instead of constructing a string and then converting to the `address` type, why not constructing the `address` record directly?

Reply:
let get_virtual_pci_address ~__context vgpu =
    let open Pci in
    let device = vgpu.Db_actions.vGPU_device in
    {
      domain = 0000;
      bus = int_of_string device;
      dev = 0;
      fn = 0;
    }
Updated like above. Thanks, Rob,

Reply:
Probably `bus = 0; dev = int_of_string device`?

Reply:
:) Yes, the code pushed is right. I made a mistake here :)